src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1138,6 +1138,61 @@ public TimeSpan ErrorEventsHarvestCycle
         public virtual uint ErrorsMaximumPerPeriod { get { return 20; } }
         public virtual IEnumerable<string> ExceptionsToIgnore { get { return ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorErrorsToIgnore, _localConfiguration.errorCollector.ignoreErrors.exception); } }
 
+        private ReadOnlyDictionary<string, IEnumerable<string>> ParseExpectedErrorConfigurations()
+        {
+            var localExpectedMessages = new Dictionary<string, IEnumerable<string>>();
+
+            foreach (var errorClass in _localConfiguration.errorCollector.expectedMessages)
+            {
+                var messages = errorClass.message;
+                if (messages != null)
+                {
+                    localExpectedMessages.Add(errorClass.name, messages);
+                }
+            }
+
+            var expectedMessages =  ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorExpectedMessages, localExpectedMessages).ToDictionary();
+
+            var expectedClasses = ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorExpectedClasses, _localConfiguration.errorCollector.expectedClasses.errorClass);
+
+            foreach (var className in expectedClasses)
+            {
+                if (expectedClasses.Contains(className))
+                {
+                    expectedMessages[className] = null;
+                }
+            }
+
+            return new ReadOnlyDictionary<string, IEnumerable<string>>(expectedMessages);
+        }
+
+        private ReadOnlyDictionary<string, IEnumerable<string>> _expectedMessages;
+        public virtual ReadOnlyDictionary<string, IEnumerable<string>> ExpectedMessages
+        {
+            get
+            {
+                if (_expectedMessages == null)
+                {
+                    _expectedMessages = ParseExpectedErrorConfigurations();
+                }
+
+                return _expectedMessages;
+            }
+        }
+
+        public virtual IEnumerable<string> ExpectedStatusCodes
```

Comment:
Similarly to the `ExpectedMessages` config item, we should consider changing this name so that it's clear that it's used for an expected error configuration.

Reply:
Ultimately the `ExpectedErrorInfo` will also contains expected error status code, but I like address this in a different PR so that I can make sure it is the right thing to do.